tls/s2n_signature_algorithms.c

```diff
@@ -50,13 +52,25 @@ static int s2n_signature_scheme_valid_to_offer(struct s2n_connection *conn, cons
 static int s2n_signature_scheme_valid_to_accept(struct s2n_connection *conn, const struct s2n_signature_scheme *scheme)
 {
     POSIX_ENSURE_REF(scheme);
+    POSIX_ENSURE_REF(conn);
 
     POSIX_GUARD(s2n_signature_scheme_valid_to_offer(conn, scheme));
 
     if (scheme->maximum_protocol_version != S2N_UNKNOWN_PROTOCOL_VERSION) {
         POSIX_ENSURE_LTE(conn->actual_protocol_version, scheme->maximum_protocol_version);
     }
 
+    POSIX_ENSURE_NE(conn->actual_protocol_version, S2N_UNKNOWN_PROTOCOL_VERSION);
+
+    if (conn->actual_protocol_version >= S2N_TLS13) {
```

Comment:
Per your comment in the PR description, I wonder if these checks make more sense in `s2n_signature_scheme_valid_to_offer`? `s2n_signature_scheme_valid_to_accept` calls `s2n_signature_scheme_valid_to_offer`, so the checks would be triggered in both functions. If the goal is to eventually replace the signature min/max versions, it seems like you might ultimately want the check in `s2n_signature_scheme_valid_to_offer`? Not sure.

Reply:
I tried that before and it failed `s2n_signature_algorithms_test`. Although looking at that test again it looks like there may be a logical issue with the test. Looking into it now.

Reply:
Okay spoke with Lindsay and clarified. The valid to offer function gets called during the client hello. It is used to generate the list of signature schemes that we offer to the peer. At this point the actual protocol version is yet to be negotiated. Which means that we can offer both TLS1.3 and TLS1.2 (and below) algorithms. Once the peer responds we will know the actual protocol version. If the newly negotiated actual protocol version is TLS1.3 then we have to reject particular signature algorithms (sha1, rsa) even if we previously offered them.

Reply:
These checks rely on knowing the actual protocol version. Therefore they must go in valid to accept and cannot go into valid to offer.

Reply:
Ahh got it okay. Thanks!